midterm-2023-11-06/code/5115-nikola-tosic/script.ts

```diff
@@ -0,0 +1,162 @@
+interface Album {
+    name: string;
+    year: number;
+}
+
+interface Band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    country: string;
+    members: string[];
+    albums: Album[];
+}
+
+type BandSorter = (first: Band, second: Band) => number;
+
+document.addEventListener("DOMContentLoaded", siteCode);
+
+let bands: Band[] = [];
+
+async function siteCode() {
+    const data = await loadData();
+    bands = data;
+
+    fillCountries(bands);
+
+    displayBands(bands);
+
+    const nameSort = document.getElementById("sort-name")!;
```

Comment:
Attaching handlers for the sort and filter buttons +5(even though these buttons don't actually exist in the HTML)